src/common/types/types.cpp

```diff
@@ -1686,7 +1710,14 @@ bool LogicalTypeUtils::tryGetMaxLogicalType(const LogicalType& left, const Logic
     if (!tryGetMaxLogicalTypeID(left.typeID, right.typeID, resultID)) {
         return false;
     }
-    result = LogicalType(resultID);
+    // attempt to make complete types first
+    if (resultID == left.typeID) {
```

Comment:
Mainly when it comes to decimal, rather than constructing an incomplete LogicalType based purely on the resultID, it's more correct to construct a LogicalType by copying the LogicalType of either `left` or `right` to make a complete type.